### ISSUE TITLE
need update Python version (>= 3.12) for future LALSuite and BBHx

### ISSUE THREAD

**Issue by WuShichao** [gwastro/pycbc#4673]
(1) `LALSuite` doesn't support python3.12 right now; (2) `BBHx` wants to require python >= 3.12.
The temporary fix for BBHx is here https://github.com/gwastro/pycbc/pull/4672

**Comment by titodalcanton** [comment]
lalsuite 7.23 was released a couple days ago with Python 3.12 support. For BBHx, see the discussion at https://github.com/mikekatz04/BBHx/issues/16.

**Comment by WuShichao** [comment]
> lalsuite 7.23 was released a couple days ago with Python 3.12 support. For BBHx, see the discussion at [mikekatz04/BBHx#16](https://github.com/mikekatz04/BBHx/issues/16).

Yeah, when the BBHx issues are fixed, we can close this Python 3.12 issue.
